### PR TITLE
core: Add N(naming) ruff rules

### PR DESCRIPTION
**Pull request by cbornet**
Public classes/functions are not renamed and rule is ignored for them.

### CODE REVIEW

File: libs/core/langchain_core/output_parsers/xml.py [hunk 1]
```diff
@@ -188,17 +188,17 @@ def parse(self, text: str) -> Dict[str, Union[str, List[Any]]]:
...
+                import defusedxml  # type: ignore
             except ImportError:
                 raise ImportError(
                     "defusedxml is not installed. "
                     "Please install it to use the defusedxml parser."
                     "You can install it with `pip install defusedxml`"
                     "See https://github.com/tiran/defusedxml for more details"
                 )
-            _ET = DET  # Use the defusedxml parser
+            _et = defusedxml.ElementTree  # Use the defusedxml parser
```
Comment (cbornet) [inline libs/core/langchain_core/output_parsers/xml.py:199]: This variable doesn't seem to be used...

**Comment by eyurtsev** [comment]
@cbornet how difficult was this PR to generate? Is it relying on any auto-fixing behavior from Ruff? I'm worried about merge conflicts for 0.3 release since we're maintaining a separate branch for it and there may be merge conflicts

**Comment by eyurtsev** [comment]
@cbornet I'll see if I can merge this one or not. And we can do other linting changes once 0.3 is out

**Comment by cbornet** [comment]
@eyurtsev I've rebased after 0.3 merge.
Can you take a look ?

**Comment by eyurtsev** [comment]
@cbornet hi it looks good -- but it needs another remerge since it picked up merge conflict from the other ruff change PR that was merged yesterday

**Review by eyurtsev — APPROVED** [review]

File: libs/core/langchain_core/output_parsers/xml.py (hunk 1 above)
Comment (efriis) [reply libs/core/langchain_core/output_parsers/xml.py:200]: @eyurtsev should this be used below in the `fromstring`?

**Pull request merged**